### ISSUE TITLE
Cannot convert to JSONObject

### ISSUE THREAD

**Issue by akovacevic** [erizet/SignalA#51]
When receiving a message from SignalR, i get the following error:

686-686/com.example.adis.shuttleservice E/TransportHelper﹕ Error when calling setMessage on connection.
    org.json.JSONException: Value hello at 0 of type java.lang.String cannot be converted to JSONObject

**Comment by gxshao** [comment]
did you guys solved it ? i am stuck right here,looking for some help..thanks.

**Comment by MrshaoGx** [comment]
may be just escape character missed,let me try again..

**Comment by gxshao** [comment]
Oh..i checked the source code,and i found that JSONArray (var: _messagesArray=response.optJSONArray ()_) doesn't work properly....the messagesArray have no JSONObject but String.....

this is before:
![before](https://user-images.githubusercontent.com/15712308/30152071-807f0792-93e4-11e7-96a8-4978d1ca3735.png)

after i changed value of messageArray:
![after](https://user-images.githubusercontent.com/15712308/30152093-9a5b06b6-93e4-11e7-9e73-a6f327e44050.png)

am i right?

**Comment by gxshao** [comment]
i was solved this problem yesterday..and i committed it to my own repo.Add a for loop after L64,Tansform the  String to JSONObject before calling it.
